### DISCUSSION TITLE
API Command to grab screenshot

### DISCUSSION THREAD

**Discussion by matthewdias** [awawa-dev/HyperHDR#878] [Ideas]
Hi, would love a way to leverage the connected video capture device to take a screenshot of what im watching (and save to file), without having to kill HyperHDR. Alternatively, if this is possible with existing features, can you please share pointers on how I could achieve this?

I saw there was a similar discussion and the result was the color average command, but I'm hoping to instead get an image file so I can remember a moment or send to a friend.

Thanks for this great software.

**Comment by awawa-dev** [comment]
Hi
I recommend using the screenshot option from the interface. As for the API, due to privacy issues, it is not easy and requires administrator rights in HyperHDR, i.e. you must be logged in to receive the MJPEG stream as binary data in the websocket sesssion. I am even planning to add an option to builds that will block streaming after recompilation, because it is usually not needed when the setup is already running and limits the possibility of hacking the system, e.g. when the user did not change the default password despite the recommendations.

**Reply by matthewdias** [reply]
makes sense, thanks. I may look into modifying the frontend feed to get a higher quality preview

**Reply by awawa-dev** [reply]
probably you need to modify this method: https://github.com/awawa-dev/HyperHDR/blob/3003f8fc7e7821c8ae3f10cab20c14f3ec385548/sources/hyperimage/HyperImage.cpp#L406
